api/v1beta1/openstackprovisionserver_types.go

```diff
@@ -17,7 +17,10 @@ limitations under the License.
 package v1beta1
 
 import (
+	"context"
+
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goClient "sigs.k8s.io/controller-runtime/pkg/client"
```

Comment:
The client is imported and used here because the added functions are needed by both the webhook and the controllers.  To avoid import cycles and also allow the operator to run locally (without webhooks, and therefore without `webhookClient`, passing a client seemed like a reasonable option.  We can change this is someone has a better idea.